spyder/plugins/ipythonconsole/widgets/shell.py

```diff
@@ -470,24 +475,17 @@ def get_cwd(self):
         """
         return self._cwd
 
-    def update_cwd(self):
-        """
-        Update working directory in Spyder after getting its value from the
-        kernel.
-        """
-        if not self.spyder_kernel_ready:
-            # Frontend sends first
-            return
-        self.call_kernel(callback=self.on_getting_cwd).get_cwd()
-
-    def on_getting_cwd(self, cwd):
+    def update_state(self, state):
         """
-        If necessary, notify that the working directory was changed to other
-        plugins.
+        New state recieved from kernel.
         """
-        if cwd != self._cwd:
+        cwd = state.pop("cwd", None)
+        if cwd and self._cwd and cwd != self._cwd:
+            # Only set it self._cwd is already set
```

Comment:
```suggestion
            # Only set it if self._cwd is already set
```

I think that's what you meant here, right?